### PR TITLE
feat(real-time): add real-time package

### PR DESCRIPTION
**Pull request by nicojs**
Add real-time package. It is a helper package to help with the nodejs side of a real-time report. It abstracts away all the nitty gritty server-send events stuff underneath.

### CODE REVIEW

File: tsconfig.settings.json
Comment (hugo-vrijswijk) [inline tsconfig.settings.json:6]: Isn't this already implied with `target: es2022`?

File: packages/real-time/test/unit/mutation-event-sender.spec.ts
Comment (hugo-vrijswijk) [inline packages/real-time/test/unit/mutation-event-sender.spec.ts:14]: It seems a bit strange to me to make a stub instance for something like a http response. Is it not possible to make the object ourselves and spy on it? Especially since the sut is doing quite a lot of side-effect calling on the response this doesn't give me faith that it is being used properly

File: .vscode/launch.json
Comment (hugo-vrijswijk) [inline .vscode/launch.json:20]: 🎉

**Review by xandervedder — APPROVED** [review]
Looking forward to use this in `stryker-dashboard`! 🥳

File: tsconfig.settings.json
Comment (nicojs) [reply tsconfig.settings.json:6]: No. `"target": "es2022"` implies `"lib": ["es2022", "dom", "dom.iterable"]`. 🤷‍♀️

File: packages/real-time/test/unit/mutation-event-sender.spec.ts
Comment (nicojs) [reply packages/real-time/test/unit/mutation-event-sender.spec.ts:14]: It is a unit test. I want to avoid creating an _actual_ response. 

Do you mean to say it's better to write a custom fake that implements the `ServerResponse` interface?

**Comment by nicojs** [comment]
@hugo-vrijswijk I've commented on both remarks you've made. Could you take a look and answer my questions?

File: tsconfig.settings.json
Comment (hugo-vrijswijk) [reply tsconfig.settings.json:6]: Of course 🤷

File: packages/real-time/test/unit/mutation-event-sender.spec.ts
Comment (hugo-vrijswijk) [reply packages/real-time/test/unit/mutation-event-sender.spec.ts:14]: Well, I expected `ServerResponse` to just be a data object carrying some stream objects or something. That way you could test the streams. If that's not the case, this might be the simplest approach. Then a custom fake doesn't have much advantage over a stub

Comment (hugo-vrijswijk) [reply packages/real-time/test/unit/mutation-event-sender.spec.ts:14]: I clearly expected too much 🙃

Comment (nicojs) [reply packages/real-time/test/unit/mutation-event-sender.spec.ts:14]: Haven't you learned by now? This is JavaScript after all 😉

**Review by hugo-vrijswijk — APPROVED** [review]
@nicojs not sure if you were waiting for specific approval from my side but this looks good 👍

**Pull request merged**
